specs/experimental/plasma.md

```diff
@@ -83,6 +83,16 @@ on the `commitment_type_byte` where [0, 128) are reserved for official implement
 | `commitment_type` | `commitment`                    |
 | ----------------- | ------------------------------- |
 | 0                 | `keccak256(tx_payload)`         |
+| 1                 | `da-server`                     |
+
+The `da-server` commitment is as follows: `version_byte ++ len(payload) as uvarint ++ payload`. The version byte
```

Comment:
rather than `da-server`, having something like `da-service` makes things more clear ? Server sounds a bit like a centralised storage place, which more DA layers are not exclusively.

Reply:
There should also be a flag introduced alongside `usePlasma` or `usePlasma` should be extended to take which commitment type the system should follow. Based on that the Plasma module is initialised, so that there are no issues in op-node integration with the module.

Reply:
Should the configuration live in the DA server? It seems like we need some rethinking of how the configuration should happen

Reply:
I'm very open to renaming it from `da-server` to `da-service` or something else. The DA server does need to be able to parse the commitment & needs to handle that configuration

Reply:
I think we should, setup of DA commitment should be encapsulated in `da-server` ( or `da-service` ) .

Reply:
Just to expand on that, different da will have different payload structure, and ideally the 'GetInput' and 'Set Input' in plasma should be capable of of calling them by having an initiasation step for the different payload structures. That should be the ideal design imo, and currently its easy todo, by adding checks during new plasma type creation simply where the returned type adheres to the 'DAStorage' interface

Reply:
>Should the configuration live in the DA server? It seems like we need some rethinking of how the configuration should happen

It would be ideal if `plasma pkg` implemented a plugin architecture. Each service can provide it's custom flags for configuration which would then be used to initialise the service.

>( how to pass which DA it is ? - in here i introduced UseAvailDA , but we should look at better design patterns here )

We can add a plasma flag that specifies which plugin to use which would then initialise the plugin with its custom flags and use it for the da service.

We can look at popular ORM implementations for inspiration as this sounds like a very similar problem.

Reply:
is there any decision made on this ?

Reply:
I think there is some confusion on the location of the DA server. The da-server uses the interface `get/set`. The exact interface may need to change a little bit as we pull in different layers, but op-node should only ever need the DA server URL